### ISSUE TITLE
Use default `root` logger in `VerbosityOption`

### ISSUE THREAD

**Issue by rasa** [kdeldycke/click-extra#318]
Using click-extra 3.4.0, and modifying the second script at https://kdeldycke.github.io/click-extra/commands.html#default-options:
```python
import logging
from click_extra import extra_command, VersionOption, ConfigOption, VerbosityOption
@extra_command(params=[
   VersionOption(version="0.1"),
   ConfigOption(default="ex.yml"),
   VerbosityOption(default="DEBUG"),
])
def cli():
  logging.warning("logging at warning level")
  logging.info("logging at info level")
  logging.debug("logging at debug level")

cli()
```
produces:
```text
debug: Verbosity set to DEBUG.
debug: issue318.py, version 0.1
WARNING:root:logging at warning level
```
which is unexpected. Perhaps the docs should be updated to pass `logging.getLogger()` to the `VerbosityOption()` function:
```python
import logging
from click_extra import extra_command, VersionOption, ConfigOption, VerbosityOption
@extra_command(params=[
   VersionOption(version="0.1"),
   ConfigOption(default="ex.yml"),
   VerbosityOption(logging.getLogger(), default="DEBUG"),
])
def cli():
  logging.warning("logging at warning level")
  logging.info("logging at info level")
  logging.debug("logging at debug level")

cli()
```
which outputs the expected result:
```text
WARNING:root:logging at warning level
INFO:root:logging at info level
DEBUG:root:logging at debug level
```
Also, notice that the first two lines of the output with the `debug` prefix do not appear with this second script. It appears click-extra has its own logger. This most likely will confuse the user (as it's confusing to me).

**Comment by kdeldycke** [comment]
I agree, the situation about loggers is quite confusing. Mainly due to the fact that I tried to integrate with the `click-log` module. But I just removed that dependency in the development branch.

In the past few weeks I also refactored the logging utilities of Click Extra to clear out all the mess:
- Click Extra now has its own internal `click-extra` logger
- The `--verbosity` option now defaults to Python's `root` logger
- You can pass your own logger object or logger name string to `--verbosity` (in which case `logging.getLogger()` will be called)

So all your concerned have now been addressed in the upcoming Click Extra v4.0.0. The major version upgrade is due to the big refactor, the drop of Python 3.7 and other additional features.

The new documentation also mirrors some of you own examples. Check it out at: https://kdeldycke.github.io/click-extra/logging.html

I have some edge-cases and flaky tests to fix before releasing Click Extra v4.0.0, but you can start to play with the `main` branch in the mean time.

**Comment by kdeldycke** [comment]
This issue has been fixed by: https://github.com/kdeldycke/click-extra/commit/d9776e2fe72917dd74208dcc824cc1a6eec2f650
